### PR TITLE
Module Editor Infrastructure + Chakra Design System

### PR DESCRIPTION
**Pull request by draymond63**
## Implementation Description
Two major details implemented in this PR. The structure of the module editor (as discussed in the architecture meeting) is laid out. There is no data retrieval, or much visual stuff at all. The main feature I built out was the SideBar, but that will be improved by someone in a future ticket most likely.

## Screenshots
image

## What should reviewers focus on?
- Do all the files make sense? Should things be renamed/moved?
- Check the live link at https://rowan-house-society--pr74-module-editor-7xowzrnn.web.app/admin/edit-module/exampleName (you'll have to login/signup first)
- Play with the tabs in the sidebar that don't look like tabs to make sure they work (they definitely should)

## Checklist
- [x] Ensure code follows style guide by running the linters
- [ ] I have updated the documentation or deemed it unnecessary
- [ ] I have linked the relevant issue in this PR
- [x] I have requested a review from the PL, as well as other dev(s) (and designers if necessary)

### CODE REVIEW

File: frontend/src/components/common/content/ImageBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { ContentImageProps } from "../../../types/ModuleEditorTypes";
+
+const TextBlock = ({
+  content,
+}: {
+  content: ContentImageProps;
+}): React.ReactElement => {
+  return <img src={content.link} alt="We should put an alt" />;
```
Comment (e-wai) [inline frontend/src/components/common/content/ImageBlock.tsx:9]: Do we want captions in images? Don't think they were in any of the designs

File: frontend/src/components/module-editor/BlockPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { ContentType } from "../../types/ModuleEditorTypes";
+
+const BlockPreview = ({ type }: { type: ContentType }): React.ReactElement => {
+  let title = "";
+
+  switch (type) {
+    case ContentType.text:
```
Comment (e-wai) [inline frontend/src/components/module-editor/BlockPreview.tsx:8]: Thoughts on giving the ContentType enum a text string name? Whether that's an attribute or the auto-generated name

File: frontend/src/components/module-editor/LessonViewer.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import { EditorContext } from "../../contexts/ModuleEditorContext";
+import EditableContentBlock from "./EditableContentBlock";
+
+const LessonViewer = (): React.ReactElement => {
+  const editorContext = useContext(EditorContext);
+
+  return (
+    <VStack flex="1" flexFlow="column" alignItems="center" spacing="8">
+      {editorContext.lesson.content.map((x, i) => (
```
Comment (e-wai) [inline frontend/src/components/module-editor/LessonViewer.tsx:12]: Can we use more specific var names for `x, i`?

File: frontend/src/components/module-editor/SideBarContentKiosk.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { ContentType } from "../../types/ModuleEditorTypes";
+
+import BlockPreview from "./BlockPreview";
+
+const SideBarContentKiosk = (): React.ReactElement => {
+  const options = [ContentType.text, ContentType.image];
+
+  return (
+    <div>
+      {options.map((x, i) => (
```
Comment (e-wai) [inline frontend/src/components/module-editor/SideBarContentKiosk.tsx:11]: more helpful names for x, i vars?

File: frontend/src/theme/components/button.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
+const Button = {
```
Comment (e-wai) [inline frontend/src/theme/components/button.ts:1]: Might be helpful to document this (unless it's already somewhere I'm missing)

(eg. gif that shows what each button looks like + interaction)

**Review by e-wai — COMMENTED** [review]
Only did a really quick readthrough

File: frontend/src/components/common/content/ImageBlock.tsx (hunk 1 above)
Comment (draymond63) [reply frontend/src/components/common/content/ImageBlock.tsx:9]: Good question. These content blocks will require more work, but I'll definitely ask design about this.

File: frontend/src/theme/components/button.ts (hunk 5 above)
Comment (draymond63) [reply frontend/src/theme/components/button.ts:1]: That's kind of documented here, perhaps I could take screenshots and put it in Notion? https://www.figma.com/file/zh6zk2LJo0RodIHBVycylC/RHS-Design-System?node-id=209%3A13640

File: frontend/src/components/module-editor/BlockPreview.tsx (hunk 2 above)
Comment (draymond63) [reply frontend/src/components/module-editor/BlockPreview.tsx:8]: Do you mean storing the shown name as the value of the enum? If so, I'm unsure how that would scale if we also wanted to add preview images to each thing. I was thinking the switch case was the most versitile.

File: frontend/src/theme/components/button.ts (hunk 5 above)
Comment (draymond63) [reply frontend/src/theme/components/button.ts:1]: I'll create a Notion page outlining the possible variants.

File: frontend/src/components/module-editor/BlockPreview.tsx (hunk 2 above)
Comment (draymond63) [reply frontend/src/components/module-editor/BlockPreview.tsx:8]: Switching to a class based implementation.

**Comment by draymond63** [comment]
Merging after conversation from meeting today.

**Pull request merged**